### DISCUSSION TITLE
能支持wolai吗？

### DISCUSSION THREAD

**Discussion by xlenco** [LetTTGACO/elog#65] [Ideas]
https://www.wolai.com/wolai/7FB9PLeqZ1ni9FfD11WuUi
开发者文档

**Comment by LetTTGACO** [comment]
看起来API 是完全支持的，我抽空研究下

**Comment by LetTTGACO** [comment]
https://www.wolai.com/wolai/8vB1m96e4QYiRzjStcaen5
<img width="864" alt="image" src="https://github.com/LetTTGACO/elog/assets/37357188/9ef318c1-9c61-44ef-bf82-313fd67c220a">


但是这个 API 用量限制 貌似免费版完全不够用...

**Reply by xlenco** [reply]
🤣没看到，耽误开发者大大时间了，它被阿里收购了，不知道以后会不会改。

**Comment by xlenco** [comment]
不过如果每调去一次可以导出所有文件，那每月次数感觉还是够的。

**Reply by LetTTGACO** [reply]
这个次数不是指的 elog 导出次数，而是 Elog 中调用 wolai 的 API 次数。一般来说导出的流程如下：
 1. 获取表格文档列表，不含详情
 2. 获取单篇文档，含详情，有多少篇文档就会调用多少次
 3. 将文档详情数据转为 md 格式字符串，导出到本地
这里面调用次数最多的就是获取单篇文档的内容，如果限制为每小时 10 次，那么相当于，一个小时内只能下载 10 篇以下的文档🤣
我看了下他的 API，貌似还没有直接获取单篇文档的全部内容，还需要一个块一个块的获取，这也是 api 调用。

如果真限制为 10 次/小时，那真的有点耍猴了。当然这只是我初步判断，得等周末我测试一下。

**Reply by xlenco** [reply]
看着钉钉个人版中 wolai 的容量，我彻底放弃了我的幻想
![Screenshot_20240222_211355_edit_507661662127435.jpg](https://github.com/LetTTGACO/elog/assets/75837117/e4632b24-edae-4639-8461-6ca578287817)

**Comment by LetTTGACO** [comment]
找到一个取消救国，不用官方提供的 Token，而是用他自己的导出功能 调用时cookie 中的 Token
在程序中复现，来调用他自己的导出接口，目前试了下是可行的
会返回单篇文档的 md网络文件路径，然后可以重新请求这个 URL 拿到 Md 字符串
<img width="717" alt="image" src="https://github.com/LetTTGACO/elog/assets/37357188/43c8af72-2b96-4453-b249-66ab416ddaab">

周末我先试一试

**Comment by LetTTGACO** [comment]
我没有用他提供的API 文档来导出 md，我使用了网页端自带的导出接口，模拟了导出 api，目前是可行的，而且不受 API 调用限制
<img width="620" alt="image" src="https://github.com/LetTTGACO/elog/assets/37357188/c4311514-cf39-4ee9-8e44-f665ab81c34a">
目前基本功能已经可以了，预计会在下个月上线

**Comment by LetTTGACO** [comment]
https://github.com/LetTTGACO/elog/releases/tag/v0.14.0-beta.0

欢迎测试，如有需要，请邮箱联系我：me@mail.1874.cool，拉你进用户群交流
